pcdsdevices/utils.py

```diff
@@ -156,3 +158,47 @@ def get_component(obj):
         return None
 
     return getattr(type(obj.parent), obj.attr_name, None)
+
+
+def schedule_task(func, args=None, kwargs=None, delay=None):
+    """
+    Use ophyd's dispatcher to schedule a task for later.
+
+    This is basically the function I was hoping to find in ophyd.
+    Schedules a task for the utility thread if we're in some arbitrary thread,
+    schedules a task for the same thread if we're in one of ophyd's callback
+    queues already.
+    """
```

Comment:
@klauer this is my answer to the dispatcher usage based on your information/examples. I'd appreciate it if you could give it a once-over tomorrow because it scares me a little (though it definitely works for the imagers).